src/differentiation/jaches_products.jl

```diff
@@ -207,19 +207,23 @@ struct FwdModeAutoDiffVecProd{F,U,C,V,V!} <: AbstractAutoDiffVecProd
 end
 
 function update_coefficients(L::FwdModeAutoDiffVecProd, u, p, t)
-    FwdModeAutoDiffVecProd(L.f, u, L.vecprod, L.vecprod!, L.cache)
+    f = update_coefficients(L.f, u, p, t)
+    FwdModeAutoDiffVecProd(f, u, L.cache, L.vecprod, L.vecprod!)
 end
 
 function update_coefficients!(L::FwdModeAutoDiffVecProd, u, p, t)
+    update_coefficients!(L.f, u, p, t)
     copy!(L.u, u)
     L
 end
 
 function (L::FwdModeAutoDiffVecProd)(v, p, t)
+    L = update_coefficients(L, v, p, t)
```

Comment:
Is this correct? We want to do f'(u)*v, so the operator's values would be `u`, not `v`. We only need to update coefficients when `u` changes, as the operator itself shouldn't change when evaluating with a new `v`.

Reply:
> I think my mental model of op(u,p,t) right now is:

But this is op(v,p,t) where the `v` is different from `u`.

> If it were just op * u, we wouldn't need the p and t parameters in the signature either, right, and just do op(u)? So if we're accepting the p and t here, it only makes sense if we're doing an update coefficients. For f'(u) v, we would do update_coefficients!(op, u, p, t) and then op * v. (And that's how e.g. ODE.jl should use the Jacobian, with the *)

`p` and `t` need to be there for if it's used as the ODE definition. In that case, `u' = Lu` with a JacVec operator would implicitly be solving `u' = f'(u)*u` for this reason. 

It might even make sense to just remove the `L(u,p,t)` operations in the near future as they weren't found to be all that useful due to this limitation.

Reply:
Your original comment is resolved: lines are removed, and tests have been updated so that the would not pass if they were there.